tests/check.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+#include "saiattributelist.h"
+
+struct Check {
+    static bool AttrListEq(sai_object_type_t objecttype, const std::vector<sai_attribute_t>& act_attr_list, /*const*/ SaiAttributeList& exp_attr_list)
+    {
+        if (act_attr_list.size() != exp_attr_list.get_attr_count()) {
+            return false;
+        }
+
+        for (uint32_t i = 0; i < exp_attr_list.get_attr_count(); ++i) {
+            sai_attr_id_t id = exp_attr_list.get_attr_list()[i].id;
+            auto meta = sai_metadata_get_attr_metadata(objecttype, id);
+
+            assert(meta != nullptr);
```

Comment:
>assert [](start = 12, length = 6)

'assert' will be nothing in non debug build and you skip the check. Is it your intent?

Reply:
Because this `meta` query from expectation value. So it shall not happen in any time except developing phase.